src/peergos/server/messages/ServerMessageStore.java

```diff
@@ -0,0 +1,175 @@
+package peergos.server.messages;
+
+import peergos.server.sql.*;
+import peergos.server.util.Logging;
+import peergos.shared.cbor.*;
+import peergos.shared.corenode.*;
+import peergos.shared.crypto.asymmetric.*;
+import peergos.shared.crypto.hash.*;
+import peergos.shared.storage.*;
+import peergos.shared.user.*;
+import peergos.shared.util.*;
+
+import java.sql.*;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.function.*;
+import java.util.logging.*;
+import java.util.stream.*;
+
+public class ServerMessageStore implements ServerMessager {
+    private static final Logger LOG = Logging.LOG();
+
+    private static final String SELECT = "SELECT id, type, sent, body, priorid, dismissed FROM messages WHERE username = ?;";
```

Comment:
SQL is good at sorting. If returned in chronological order from DB there is less work to do in Java code later.

Reply:
I prefer to keep it simple, and not rely on the db for too much. This results is greater portability across databases.